docker-compose.yml

```
@@ -1601,19 +1601,19 @@ services:
       /bin/bash -c "/arrow/ci/scripts/r_test.sh /arrow"
 
   ubuntu-r-sanitizer:
-    # Only 20.04 and amd64 supported
+    # Only 22.04 and amd64 supported
     # Usage:
     #   docker compose build ubuntu-r-sanitizer
     #   docker compose run ubuntu-r-sanitizer
-    image: ${REPO}:amd64-ubuntu-20.04-r-sanitizer
+    image: ${REPO}:amd64-ubuntu-22.04-r-sanitizer
```

```suggestion
    image: ${REPO}:amd64-ubuntu-r-sanitizer
```

I'm fine keeping 22.04 here, but if we can't really control what Winston's image is using, I wonder if we need to specify the version here (and then below) at all? Then it's "just" updating that one comment when we do this next?

---

You're right. Let's do it.

---

I also removed Ubuntu version in comment because we don't need the information here. (We can't customize it by environment variable like other service such as `UBUNTU=24.04 archery docker run ubuntu-cpp`.)